### DISCUSSION TITLE
Is there a way to use {tokens} in the SearchPageUrl?

### DISCUSSION THREAD

**Discussion by mozilla0** [microsoft-search/pnp-modern-search#2954] [Q&A]
We try to make a dynamic master search result page which is pre-filtered from the search request of a source page. The idea is to use tokens in the url (based on PropertyBags of the source site collection).

So the url is setup with: `Set-PnPSearchSettings -SearchPageUrl "/sites/search/SitePages/results.aspx?jit={Request.Url}&jit2=BusinessLocation1"`

![2023-05-05_100116](https://user-images.githubusercontent.com/17600172/236406417-73997c66-b57f-408b-9abe-f51c076ccf01.png)

The result looks like these:
![2023-05-05_100216](https://user-images.githubusercontent.com/17600172/236406435-33ab0d7a-b425-4bc5-a306-083f111330e3.png)
![2023-05-05_100259](https://user-images.githubusercontent.com/17600172/236406445-2fbd584f-7a2e-4e50-89a2-1dc3446b4964.png)

But the tokens are not resolved. Is there any way to make this SearchPageUrl dynamic?

**Comment by kasperbolarsen** [comment]
Hi Matthias, have you looked at the dynamic option in the search result page? https://microsoft-search.github.io/pnp-modern-search/usage/search-results/connections/

Or the options in the search box https://microsoft-search.github.io/pnp-modern-search/usage/search-box/

**Comment by mozilla0** [comment]
Yes, but that’s only the way for using the tokens . My problem is that we have to generate the link to search center dynamically: means the source site collection (which is another than the site collection of the search center) has to bring individual parameters (as tokens) to the searchPage (and use it there with the dynamic option). My idea was to use tokens, but it seems that SharePoint ignore this tokens if the are set from the PnPSearchSettings.
